controllers/admin/AdminCarriersController.php

```diff
@@ -649,6 +649,27 @@ public function getList($id_lang, $order_by = null, $order_way = null, $start =
                 $this->_list[$key]['name'] = Carrier::getCarrierNameFromShopName();
             }
         }
+
+        // Add specific sort for naming
+        if ($this->_orderBy == 'name') {
+            usort($this->_list, [$this, 'sortListByCarrierName']);
+        }
+    }
+
+    /**
+     * Sort arrays based on the carrier name
+     *
+     * @param array $a The first array to compare
+     * @param array $b The second array to compare
+     *
+     * @return int 0 if they are equal. If the order way is ASC, -1 for lower and 1 for greater, otherwise, it's the opposite
+     */
+    private function sortListByCarrierName(array $a, array $b): int
+    {
+        $aName = isset($a['name']) ? strtolower($a['name']) : '';
+        $bName = isset($b['name']) ? strtolower($b['name']) : '';
+
+        return $this->_orderWay == 'ASC' ? $aName <=> $bName : $bName <=> $aName;
```

Comment:
Sorry, I should have known better. You can replace all this by the proper way to do sort by column. Here it is, for what it's worth:
```suggestion
            $names = array_column($this->_list, 'name');
            $sortOrder = $this->_orderWay == 'ASC' ? SORT_ASC : SORT_DESC;
            array_multistort($this->_list, $names, $sortOrder, SORT_NATURAL);
```

The real problem I'm seeing right now with your solution is that the sort is only performed on the current paginated result, not on the whole list.

Reply:
`usort` is really faster than `array_multisort` which is designed to compare elements from different arrays (or sub-arrays) at the same time. The `usort` is called on the `$this->_list` which is the same as you :thinking:

Reply:
I close this conversation, because I fix with a better solution.